service/controller/app/v1/resource/configmap/current.go

```diff
@@ -20,6 +21,31 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	ns, err := r.k8sClient.CoreV1().Namespaces().Get(key.Namespace(cr), metav1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	if ns.GetDeletionTimestamp() != nil {
+		currentCR, err := r.g8sClient.ApplicationV1alpha1().Apps(cr.Namespace).Get(cr.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		currentCR.Finalizers = []string{}
```

Comment:
cleaning finalizers as empty, so not to call app-operator for deletion event again.

Reply:
That fucks with all operators adding their own finalizers. So that is definitely not what we want. What a resource should do is to remove its own finalizer if there is anything like that. If there is a project wide finalizer there should be a separate resource managing it. I think when you only want to stop reconciling on deletion the resources should be canceled here, and then each should do that on their own.

Reply:
100% i discussed that with @tomahawk28 in slack. Ji is going to post in #sig-operator to discuss and find a solution that works.